### PR TITLE
Resource and data source acceptance test for provisioner resource

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**:  Resource and data source acceptance test for provisioner resource

Screenshot 2023-12-22 at 1 12 50 PM

Screenshot 2023-12-22 at 1 13 37 PM

3. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

4. **Additional information**

5. **Special notes for your reviewer**

### CODE REVIEW

File: internal/resources/provisioner/data_source_provisioner.go [hunk 1]
```diff
@@ -55,18 +55,34 @@ func dataSourceProvisionerRead(ctx context.Context, d *schema.ResourceData, m in
...
+		resp, err := config.TMCConnection.ProvisionerResourceService.ProvisionerResourceServiceList(model.FullName)
+		if err != nil {
+			if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
+				_ = schema.RemoveFromState(d, m)
+				return
+			}
+		}
+
+		for i := range resp.Provisioners {
+			d.SetId(resp.Provisioners[i].Meta.UID)
```
Comment (ramya-bangera) [inline internal/resources/provisioner/data_source_provisioner.go:68]: If the array contains more than one response will it not overwrite the ID?

File: internal/resources/provisioner/provisioner_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	provisionerName := acctest.RandomWithPrefix("tf-prv-test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: getTestProvisionerWithResourceConfigValue(provisionerName),
+				Check: resource.ComposeTestCheckFunc(
```
Comment (ramya-bangera) [inline internal/resources/provisioner/provisioner_resource_test.go:41]: Its better to add an update operation too in the test, you can create a ticket and add it later if there are time constraint

File: internal/resources/provisioner/data_source_provisioner.go (hunk 1 above)
Comment (shobha2626) [reply internal/resources/provisioner/data_source_provisioner.go:68]: The expectation is to set unique ID for every provisioner in the list

File: internal/resources/provisioner/provisioner_resource_test.go (hunk 2 above)
Comment (shobha2626) [reply internal/resources/provisioner/provisioner_resource_test.go:41]: Will take it up as part of the mock test PR or will take up later based on the time constraint

Comment (shobha2626) [reply internal/resources/provisioner/provisioner_resource_test.go:41]: Created jira - https://jira.eng.vmware.com/browse/TMC-49960

**Pull request merged**